src/java.base/share/classes/java/nio/MappedByteBuffer.java

```diff
@@ -353,4 +366,44 @@ public final MappedByteBuffer rewind() {
         super.rewind();
         return this;
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @apiNote
+     * Reading bytes into physical memory by invoking {@code load()} on the
+     * returned buffer, or writing bytes to the storage device by invoking
+     * {@code force()} on the returned buffer, will only act on the sub-range
+     * of this buffer that the returned buffer represents, namely
+     * {@code [position(),limit()]}.  A similar consideration applies also to
+     * {@code alignedSlice()} with of course differing position and length.
+     */
+    @Override
+    public abstract MappedByteBuffer slice();
+
+    /**
+     * {@inheritDoc}
+     *
+     * @apiNote
+     * Reading bytes into physical memory by invoking {@code load()} on the
+     * returned buffer, or writing bytes to the storage device by invoking
+     * {@code force()} on the returned buffer, will only act on the sub-range
+     * of this buffer that the returned buffer represents, namely
+     * {@code [position()+index,position()+index+length]}, where {@code index}
+     * and {@code length} are assumed to satisfy the preconditions.
+     */
+    @Override
+    public abstract MappedByteBuffer slice(int index, int length);
```

Comment:
The updates to make use covariant returns is good but I'm not sure about the proposal "API note"s. I should drop "API note" because it's really spec text. Also I think it would be simpler to drop the "A similar consideration ..." to avoid confusing the reader.